### ISSUE TITLE
Update mozilla-os2 to the tip of http://hg.mozilla.org/releases/mozilla-esr38/

### ISSUE THREAD

**Issue by dryeo** [bitwiseworks/mozilla-os2#172]
The bitwiseworks/mozilla-os2 tree should import the Thunderbird fixes that were committed after Firefox_38_8_0esr_RELEASE was tagged by merging in http://hg.mozilla.org/releases/mozilla-esr38/archive/tip.tar.bz2 which will pull in a couple of Thunderbird related fixes. This will keep all the Mozilla apps in sync when built with the Bitwiseworks tree.
See http://hg.mozilla.org/releases/mozilla-esr38/

**Comment by dmik** [comment]
Yes, having a single tree for both FF and TB is something that we want to do for long.
Do you want to say that updating our tree to e.g. http://hg.mozilla.org/releases/mozilla-esr38/rev/THUNDERBIRD_38_8_0_RELEASE will make Thunderbird buildable as well?

**Comment by dryeo** [comment]
Thunderbird is currently buildable, just the correct way to update to 
38.8.0esr is to pull the comm-esr38 tag for TB38.8.0 and then to cd into 
mozilla/ and pull the mozilla-esr38 tag for TB38.8.0. Of course I'm 
using the bitwise tree instead of mozilla-esr38 in mozilla/ and to build 
Firefox just means starting out in mozilla/ or using the correct 
mozconfig and running the top level configure, the mozilla/configure or 
mozilla/mach. Lately I've preferred using mach.

**Comment by dmik** [comment]
Okay, now I recall that you already told me that (or I read it myself somewhere). But this approach builds two separate XUL.DLLl & rest for TB and FF, right?

**Comment by dryeo** [comment]
Yes, that is why the readme has detailed instructions for running 
multiple Mozilla apps concurrently (LIBPATHSTRICT, which was introduced 
for this back when IBM was maintaining Mozilla or RUN! which was also 
written mainly for Mozilla).
Lately all the apps do work with the TB xul.dll (if the same version), 
Firefox xul.dll is missing an import for lightning and using the SM 
xul.dll dies with a trap in mozalloc IIRC. No guarantees that it'll 
continue to work though.
At one point the plan was to use xul-runner to run all the apps but 
Mozilla never finished it and xul-runner is now dead

**Comment by dmik** [comment]
Okay, thanks for this info. We surely want to use the same XUL.DLL for both FF and TB (and SM although I'm not sure if we will provide "official" SM builds). And even if they break it upstream we will keep it locally working as it saves a ton of memory for the end user (especially with all the other recent DLLs) .
